addons/udes_stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -1,5 +1,93 @@
-from odoo import models
+from odoo import models, fields, _, api
+from odoo.exceptions import UserError
+import logging
+
+_logger = logging.getLogger(__name__)
 
 
 class StockPickingBatch(models.Model):
     _inherit = "stock.picking.batch"
+
+    state = fields.Selection(
+        selection=[
+            ("draft", "Draft"),
+            ("waiting", "Waiting"),
+            ("ready", "Ready"),
+            ("in_progress", "Running"),
+            ("done", "Done"),
+            ("cancel", "Cancelled"),
+        ],
+        compute="_compute_state",
+        store=True,
+    )
+
+    @api.constrains("user_id")
```

Comment:
Yes, seems like a mistake...

Reply:
Could it be that who wrote it originally was also trying to change the state of the batch based on whether a user_id is assigned? I wouldn't say that's correct either but just a thought.

Reply:
I really don't know, has been like that in v11 since 2018!

For v14 I would suggest changing this to:

```
@api.depends("user_id", "picking_ids", "picking_ids.batch_id")
```

That ensures the state will be recomputed in the following cases:

* The user is changed
* The pickings are changed on the batch
* The batch is changed via the picking

Reply:
@PeterC10 Resolving the thread as is addressed to your suggestion